lib/Core/Executor.cpp

```diff
@@ -794,9 +795,21 @@ void Executor::initializeGlobals(ExecutionState &state) {
       ObjectState *wos = state.addressSpace.getWriteable(mo, os);
       
       initializeGlobalObject(state, wos, i->getInitializer(), 0);
-      // if(i->isConstant()) os->setReadOnly(true);
+      if(i->isConstant())
+        constantObjects.emplace_back(wos);
     }
   }
+
+  // TODO change workaround for current memory model
```

Comment:
Can you elaborate this TODO (or remove it), I don't know what you mean.

Reply:
I removed it for now. It's more straight forward.